pkg/rpc/context_test.go

```diff
@@ -788,453 +800,51 @@ func BenchmarkInternalClientAdapter(b *testing.B) {
+		return loopLn.Connect(ctx)
 	}
-	// The connection to Node 3 on the lisNonExistentConnection should be
-	// initializing and the server connection should be nominal.
-	testutils.SucceedsSoon(t, func() error {
-		return checkNominal(m, 1)
-	})
-
 	// Ensure that there's no error connecting to the local node when an internal
 	// server has been registered.
 	clientCtx.SetLocalInternalServer(
 		&internalServer{},
 		ServerInterceptorInfo{}, ClientInterceptorInfo{})
-	require.NoError(t, clientCtx.TestingConnHealth(clientCtx.Config.AdvertiseAddr, clientNodeID))
-
-	// Connections should shut down again and now that we're nearing the test it's
-	// a good opportunity to check that what came up must go down.
-	hbSuccess.Store(false)
-	// Need to close these or we eat a 20s timeout.
-	netutil.FatalIfUnexpected(lisNonExistentConnection.Close())
-	netutil.FatalIfUnexpected(lisNotLocalServer.Close())
-	testutils.SucceedsSoon(t, func() error {
-		started, exited := m.HeartbeatLoopsStarted.Count(), m.HeartbeatLoopsExited.Count()
-		if started != exited {
-			return errors.Errorf("started(%d) != exited(%d)", started, exited)
-		}
-		return nil
-	})
-}
 
-func checkNominal(m *Metrics, exp int64) error {
-	if n := m.HeartbeatsNominal.Value(); n != exp {
-		return errors.Errorf("%d nominal, want %d", n, exp)
-	}
-	return nil
-}
-
-// TestConnectionRemoveNodeIDZero verifies that when a connection initiated via
-// GRPCDialNode fails, we also clean up the connection returned by
-// GRPCUnvalidatedDial.
-//
-// See #37200.
-func TestConnectionRemoveNodeIDZero(t *testing.T) {
-	defer leaktest.AfterTest(t)()
-
-	ctx := context.Background()
-	stopper := stop.NewStopper()
-	defer stopper.Stop(ctx)
-
-	clock := &timeutil.DefaultTimeSource{}
-	maxOffset := time.Nanosecond
-	clientCtx := newTestContext(uuid.MakeV4(), clock, maxOffset, stopper)
-	// Provoke an error.
-	_, err := clientCtx.GRPCDialNode("127.0.0.1:notaport", 1, DefaultClass).Connect(context.Background())
-	if err == nil {
-		t.Fatal("expected some kind of error, got nil")
-	}
-
-	// NB: this takes a moment because GRPCDialRaw only gives up on the initial
-	// connection after 1s (more precisely, the redialChan gets closed only after
-	// 1s), which seems difficult to configure ad-hoc.
-	testutils.SucceedsSoon(t, func() error {
-		var keys []connKey
-		clientCtx.m.mu.RLock()
-		defer clientCtx.m.mu.RUnlock()
-		for k := range clientCtx.m.mu.m {
-			keys = append(keys, k)
-		}
-		if len(keys) > 0 {
-			return errors.Errorf("still have connections %v", keys)
-		}
-		return nil
-	})
-}
-
-type interceptingListener struct {
-	net.Listener
-	connCB func(net.Conn)
-}
-
-func (ln *interceptingListener) Accept() (net.Conn, error) {
-	conn, err := ln.Listener.Accept()
-	if err == nil {
-		ln.connCB(conn)
-	}
-	return conn, err
-}
-
-// TestHeartbeatHealth verifies that the health status changes after
-// heartbeats succeed or fail due to transport failures.
-func TestHeartbeatHealthTransport(t *testing.T) {
```

Comment:
This test is still referenced in comments in this file grpcRunKeepaliveTestCase()